### PR TITLE
Docs: Add TPC-H and TPC-DS tutorials

### PR DESCRIPTION
**Pull request by rschu1ze**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit e8f574008b48cab949b3138ae7be7863b4ebf4ba with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69877/e8f574008b48cab949b3138ae7be7863b4ebf4ba/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/69877/e8f574008b48cab949b3138ae7be7863b4ebf4ba/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

**Comment by rschu1ze** [comment]
@justindeguzman Docs Check complains and I don't understand how to fix the issue. Could you take a brief look, please? Thanks.

File: docs/en/getting-started/example-datasets/tpcds.md [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+$ clickhouse-client --format_csv_delimiter '|' --query "INSERT INTO web_returns FORMAT CSV" < web_returns.tbl
+$ clickhouse-client --format_csv_delimiter '|' --query "INSERT INTO web_sales FORMAT CSV" < web_sales.tbl
+$ clickhouse-client --format_csv_delimiter '|' --query "INSERT INTO web_site FORMAT CSV" < web_site.tbl
+```
+
+Then run the generated queries.
+
+::::warning
+TPC-DS makes heavy use of correlated subqueries which are at the time of writing (September 2024) not supported by ClickHouse ([issue #6697](https://github.com/ClickHouse/ClickHouse/issues/6697)).
+As a result, many of below benchmark queries will fail with errors.
```
Comment (nickitat) [inline docs/en/getting-started/example-datasets/tpcds.md:595]:
```suggestion
As a result, many of the benchmark queries above will fail with errors.
```

**Review by nickitat — APPROVED** [review]

File: docs/en/getting-started/example-datasets/tpcds.md [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+    web_gmt_offset        Decimal(7,2),
+    web_tax_percentage    Decimal(7,2),
+    PRIMARY KEY (web_site_sk)
+);
+```
+
+The data can be imported as follows:
+
+``` bash
+$ clickhouse-client --format_csv_delimiter '|' --query "INSERT INTO call_center FORMAT CSV" < call_center.tbl
```
[the remark this answers is not in the document]
Comment (rschu1ze) [reply docs/en/getting-started/example-datasets/tpcds.md]: True, rewrote it.

File: docs/en/getting-started/example-datasets/star-schema.md [hunk 3]
```diff
@@ -1,36 +1,36 @@
...
-Then, generate the data. Parameter `-s` specifies the scale factor. For example, with `-s 100`, 600 million rows (67 GB) are generated.
+Then, generate the data. Parameter `-s` specifies the scale factor. For example, with `-s 100`, 600 million rows are generated.
 
 ``` bash
-$ ./dbgen -s 1000 -T c
-$ ./dbgen -s 1000 -T l
-$ ./dbgen -s 1000 -T p
-$ ./dbgen -s 1000 -T s
-$ ./dbgen -s 1000 -T d
+declare -a tables=("c" "l" "p" "s" "d")
```
Comment (alexey-milovidov) [inline docs/en/getting-started/example-datasets/star-schema.md:28]: It's not clear if that's better :)

At least working with arrays in bash requires a high cognitive load.

Comment (rschu1ze) [reply docs/en/getting-started/example-datasets/star-schema.md:28]: I don't have a preference. Happy reviewers approve PRs faster :-)

Comment (alexey-milovidov) [reply docs/en/getting-started/example-datasets/star-schema.md:28]: I'm uncertain that this script is even copy-pastable. The previous script can be run step by step. Let's keep the previous.

Comment (alexey-milovidov) [inline docs/en/getting-started/example-datasets/star-schema.md:28]: Also, the usage of double quotes " is not motivated (in contrast to single quotes).

File: docs/en/getting-started/example-datasets/tpcds.md [hunk 4]
```diff
@@ -0,0 +1,578 @@
...
+# TPC-DS (2012)
+
+Similar to the [Star Schema Benchmark (SSB)](star-schema.md), TPC-DS is based on [TPC-H](tpch.md), but it took the opposite route, i.e. it expanded the number of joins needed by storing the data in a complex snowflake schema (24 instead of 8 tables).
+The data distribution is skewed (e.g. normal and Poisson distributions).
+It includes 99 reporting and ad-hoc queries with random substitutions.
+
+References
+- [The Making of TPC-DS](https://dl.acm.org/doi/10.5555/1182635.1164217) (Nambiar), 2006
+
+First, checkout the TPC-H repository and compile the data generator:
```
Comment (alexey-milovidov) [inline docs/en/getting-started/example-datasets/tpcds.md:16]: This is TPC-DS, not TPC-H.

File: docs/en/getting-started/example-datasets/tpcds.md [hunk 5]
```diff
@@ -0,0 +1,578 @@
...
+
+```bash
+./dsqgen -DIRECTORY ../query_templates/ -INPUT ../query_templates/templates.lst  -SCALE 1 # generates 99 queries in out/query_0.sql
+```
+
+Now create tables in ClickHouse.
+You can either use the original table definitions in tools/tpcds.sql or "tuned" table definitions with properly defined primary key indexes and LowCardinality-type column types where it makes sense.
+
+```sql
+create TABLE call_center(
```
Comment (alexey-milovidov) [inline docs/en/getting-started/example-datasets/tpcds.md:40]: Why create is in lowercase but TABLE in uppercase?

File: docs/en/getting-started/example-datasets/tpch.md [hunk 6]
```diff
@@ -0,0 +1,880 @@
...
+
+First, checkout the TPC-H repository and compile the data generator:
+
+``` bash
+git clone https://github.com/gregrahn/tpch-kit.git
+cd tpch-kit/dbgen
+make
+```
+
+Then, generate the data. Parameter `-s` specifies the scale factor. For example, with `-s 100`, 600 million rows are generated for LINEITEM.
```
Comment (alexey-milovidov) [inline docs/en/getting-started/example-datasets/tpch.md:27]: The table is named `lineitem`, not `LINEITEM`

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**